node/pkg/reporter/main_test.go

```diff
@@ -2,172 +2,95 @@
+		VALUES (@name, @aggregate_interval, @submit_interval) 
+		ON CONFLICT (name) DO NOTHING
+		RETURNING name, id, submit_interval, aggregate_interval;
+	`
+	DeleteConfigQuery = `DELETE FROM configs WHERE id = @id;`
+)
 
-	admin, err := utils.Setup(utils.SetupInfo{
-		Version: "",
-		Bus:     mb,
-	})
-	if err != nil {
-		return nil, nil, err
+func setup(ctx context.Context) ([]int32, error) {
+	aggregateInterval := 400
+	submitInterval15000 := 15000
+	submitInterval60000 := 60000
+
+	configs := []Config{
+		{
+			Name:              "BTC-USDT",
+			AggregateInterval: &aggregateInterval,
+			SubmitInterval:    &submitInterval15000,
+		},
+		{
+			Name:              "ETH-USDT",
+			AggregateInterval: &aggregateInterval,
+			SubmitInterval:    &submitInterval15000,
+		},
+		{
+			Name:              "BTC-KRW",
+			AggregateInterval: &aggregateInterval,
+			SubmitInterval:    &submitInterval60000,
+		},
+		{
+			Name:              "ETH-KRW",
+			AggregateInterval: &aggregateInterval,
+			SubmitInterval:    &submitInterval60000,
+		},
 	}
 
-	testItems.admin = admin
-
-	app := New()
-	testItems.app = app
-
-	tmpData, err := insertSampleData(ctx)
-	if err != nil {
-		return nil, nil, err
-	}
-	testItems.tmpData = tmpData
-
-	v1 := admin.Group("/api/v1")
-	reporter.Routes(v1)
-
-	return reporterCleanup(ctx, admin, app), testItems, nil
-
-}
-
-func reporterCleanup(ctx context.Context, admin *fiber.App, app *App) func() error {
-	return func() error {
-		err := db.QueryWithoutResult(ctx, "DELETE FROM global_aggregates;", nil)
+	var res []int32
+	for _, config := range configs {
+		insertedConfig, err := db.QueryRow[Config](ctx, InsertConfigQuery, map[string]any{"name": config.Name, "aggregate_interval": config.AggregateInterval, "submit_interval": config.SubmitInterval})
 		if err != nil {
-			return err
+			log.Error().Err(err).Msgf("error inserting config %s", config.Name)
+			return nil, err
 		}
+		res = append(res, insertedConfig.ID)
+	}
 
-		err = db.QueryWithoutResult(ctx, "DELETE FROM proofs;", nil)
-		if err != nil {
-			return err
-		}
+	return res, nil
+}
 
-		err = db.QueryWithoutResult(ctx, "DELETE FROM configs;", nil)
+func cleanUp(ctx context.Context, configIds []int32) error {
+	for _, id := range configIds {
+		_, err := db.QueryRow[any](ctx, DeleteConfigQuery, map[string]any{"id": id})
 		if err != nil {
+			log.Error().Err(err).Msgf("error deleting config %d", id)
 			return err
 		}
+	}
 
-		err = admin.Shutdown()
-		if err != nil {
-			return err
-		}
+	db.ClosePool()
 
-		// err = app.stopReporters()
-		// if err != nil {
-		// 	return err
-		// }
-		return nil
-	}
+	return nil
 }
 
 func TestMain(m *testing.M) {
+	ctx := context.Background()
 	zerolog.SetGlobalLevel(zerolog.InfoLevel)
+	configIds, err := setup(ctx)
+	if err != nil {
+		log.Fatal().Err(err).Msg("error setting up test")
+		os.Exit(1)
+	}
+
 	code := m.Run()
-	db.ClosePool()
-	db.CloseRedis()
+
+	err = cleanUp(ctx, configIds)
```

Comment:
lets call cleanUp from deferred function right after setup so that it is definitely cleaned up